Simplify Array.Copy(a, 0, b, 0, c) to Array.Copy(a, b, c)

stephentoub opened the pull request:
corefx side of https://github.com/dotnet/coreclr/pull/27641.
cc: @jkotas 

VS regex search and replace from `Array.Copy\((.+), 0, (.+), 0` to `Array.Copy($1, $2`.  The only changes I reverted were those to the tests for Array itself.

GrabYourPitchforks commented on src/System.Private.Xml/src/System/Xml/Resolvers/XmlPreloadedResolver.cs:
```diff
@@ -351,7 +351,7 @@ public void Add(Uri uri, Stream value)
                 int size = checked((int)ms.Position);
                 byte[] bytes = new byte[size];
-                Array.Copy(ms.ToArray(), 0, bytes, 0, size);
+                Array.Copy(ms.ToArray(), bytes, size);
```
Since we're already touching this code should we delete these lines 352 - 354, replacing them instead with `byte[] bytes = ms.ToArray();`? Not sure if something like that would be in scope of this PR.

stephentoub replied:
This code does seem wonky, but I have no idea what the test coverage of this looks like, so I'm going to leave it as-is.  My changes in this PR were entirely mechanical.  Thanks, though.

GrabYourPitchforks commented on src/System.Security.Cryptography.Algorithms/src/Internal/Cryptography/TripleDesImplementation.cs:
```diff
@@ -69,7 +69,7 @@ private ICryptoTransform CreateTransform(byte[] rgbKey, byte[] rgbIV, bool encry
                 // Two-Key Triple DES contains two 8-byte keys {K1}{K2} with {K1} appended to make {K1}{K2}{K1}.
                 byte[] newkey = new byte[24];
-                Array.Copy(rgbKey, 0, newkey, 0, 16);
+                Array.Copy(rgbKey, newkey, 16);
```
IMO the original code reads a bit cleaner since it's more obvious that the destination array is being written piecemeal. But honestly it's only a weakly held opinion.

GrabYourPitchforks commented on src/System.Security.Cryptography.Cng/src/System/Security/Cryptography/TripleDESCng.cs:
```diff
@@ -124,7 +124,7 @@ byte[] ICngSymmetricAlgorithm.PreprocessKey(byte[] key)
                 // Two-Key Triple DES contains two 8-byte keys {K1}{K2} with {K1} appended to make {K1}{K2}{K1}.
                 byte[] newkey = new byte[24];
-                Array.Copy(key, 0, newkey, 0, 16);
+                Array.Copy(key, newkey, 16);
```
Same here.

stephentoub committed:
Simplify Array.Copy(a, 0, b, 0, c) to Array.Copy(a, b, c)

VS regex search and replace from `Array.Copy\((.+), 0, (.+), 0` to `Array.Copy($1, $2`, then manually reviewed.  The only changes I reverted were those to the tests for Array itself.
